### ISSUE TITLE
Fix Request: The AvoidHardcodedConnectionConfig rule is too naive

### ISSUE THREAD

**Issue by anoppe** [jborgers/PMD-jPinpoint-rules#259]
The AvoidHardcodedConnectionConfig (introduced with https://github.com/jborgers/PMD-jPinpoint-rules/issues/255)  is flagging code falsely:
```java
private static final String LOGIN_TIMEOUT_PROPERTY_KEY = "loginTimeout";
```
This LoC gets flagged by the rule `AvoidHardcodedConnectionConfig` and it shouldn't, since it only contains the name of a configuration property.
I'd propose to make the rule ignore fields that comply with certain characteristics, like, the field name needs to include 'PROPERTY_KEY' or, 'CONFIG_NAME' or variations on that.
And, update the documentation on this, of course: https://github.com/jborgers/PMD-jPinpoint-rules/blob/master/docs/JavaCodePerformance.md#ibi22

**Comment by jborgers** [comment]
Should be checked for int type
